v2/bundle/spiffebundle/bundle.go

```diff
@@ -402,3 +403,28 @@ func (b *Bundle) GetJWTBundleForTrustDomain(trustDomain spiffeid.TrustDomain) (*
 
 	return b.JWTBundle(), nil
 }
+
+func (b *Bundle) Equal(other *Bundle) bool {
+	switch {
+	case b == nil || other == nil:
+		return b == other
+	case b.trustDomain != other.trustDomain:
+		return false
+	case !reflect.DeepEqual(b.refreshHint, other.refreshHint):
```

Comment:
How would you feel about replacing this use of `reflect` with a custom `refreshHintEqual(a, b time.Duration) bool` function? `reflect` has a relatively high runtime cost and IMO it's good to avoid it when reasonable.

If you are ok with that, we can do the same thing for the other fields.

The JWT key map and the X.509 root slice comparison could be part of the jwtutil and x509util packages, respectively. SPIRE has some code (in cryptoutil package) to check if public keys are equal that you can borrow.

Reply:
> How would you feel about replacing this use of reflect with a custom refreshHintEqual(a, b time.Duration) bool function? reflect has a relatively high runtime cost and IMO it's good to avoid it when reasonable.

I agree, I was about to do something like this, but then I went for the easy path to see what others think.

> The JWT key map and the X.509 root slice comparison could be part of the jwtutil and x509util packages, respectively. SPIRE has some code (in cryptoutil package) to check if public keys are equal that you can borrow.

Cool, I'll take a look, thanks!